Misc/NEWS.d/next/Core and Builtins/2024-04-26-12-08-33.gh-issue-117657.D6Zd73.rst

```diff
@@ -0,0 +1 @@
+Continuing to whack-a-mole TSAN issues found in No-GIL tests.
```

Comment:
I added the "skip news" label to the PR. I don't think we'll want individual news entries for each of the TSAN fixes -- you can remove the NEWS entry from the PR.